cloudsync/sync/state.py

```diff
@@ -1171,6 +1171,18 @@ def change(self, age):
         if not change_set:  # todo: cache the changeset
             return None
 
+        # Fill in path if needed.
+        # Some providers' events (GDrive) do not include a path. In order to minimize provider API calls
+        # in the EventManager thread, fill in missing paths here, in the SyncManager thread, instead.
+        # This must be done for the entire changeset in order for the priority sort below to work, since
+        # priority is updated when the path changes.
+        for e in change_set:
+            for side in (LOCAL, REMOTE):
+                s = e[side]
+                if not s.path and s.exists in (EXISTS, UNKNOWN) and s.changed and s.changed > s._last_gotten:
```

Comment:
Instead of calling `unconditionally_get_latest` and setting `s._last_gotten` here, it seems better to remove the part of the condition referencing `s.changed` and call `get_latest` instead, which does both of those operations already, and checks `s.changed` in a more nuanced way

Reply:
The more I think about it, the more the fact that lint OK'd this code bugs me. WTF. Did we turn off the protection against accessing protected class members globally? Jeez

Reply:
changed to use get_latest() instead.

left usage of _last_gotten and the lint rule in place, figured this was by design (allow usage in this module only, since there are no friends in python)